dbgpt/storage/knowledge_graph/graph_retriever/base.py

```diff
@@ -0,0 +1,29 @@
+"""Graph retriever base class."""
+
+import logging
+from abc import ABC, abstractmethod
+from typing import List, Optional, Any
+
+from pydantic import Field
+
+from dbgpt._private.pydantic import ConfigDict
+from dbgpt.core import Chunk
+from dbgpt.rag.index.base import IndexStoreBase, IndexStoreConfig
+from dbgpt.storage.graph_store.graph import (
+    Direction,
+    Edge,
+    Graph,
+    GraphElemType,
+    MemoryGraph,
+    Vertex,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class GraphRetrieverBase(ABC):
+    """Graph retriever base class."""
+
+    @abstractmethod
+    async def retrieve(self, text: str) -> Any:
```

Comment:
It should be declared here as:
`retrieve(self, input: Any) -> Tuple[Graph, Any]:`
* input is compatible with input in any format. The specific type is determined by the implementation class.
* The return tuple represents a graph and additional information.